lib/ansible/plugins/action/script.py

```diff
@@ -87,41 +87,47 @@ def run(self, tmp=None, task_vars=None):
             except AnsibleError as e:
                 raise AnsibleActionFail(to_native(e))
 
+            if self._play_context.check_mode:
```

Comment:
use self._task instead (play_context is not updated per loop item)

Reply:
Fixed, good catch